### PR TITLE
Resilient indexing

### PR DESCRIPTION
**Pull request by pandu-k**
* **Please check if the PR fulfills these requirements**
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- Feature

* **What is the current behavior?** (You can also link to an open issue here)
- If a document in a large add_document calls errors, then the entire call fails (rather than just the errorful doc)

* **What is the new behavior (if this is a feature change)?**
- Add_documents continues, with errored documents being indicated in the response 

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
- Perhaps: if you have a logic predicated on an error response for an add_documents call, it would no longer work
- Instead you should inspect the response of the add documents call, and see if "errors" has been set to True. You can then inspect the returned items, to see which document/s triggered created an error and why

* **Other information**:
- Ran tox

### CODE REVIEW

File: src/marqo/tensor_search/tensor_search.py [hunk 1]
```diff
@@ -329,20 +352,29 @@ def add_documents(config: Config, index_name: str, docs: List[dict], auto_refres
                     # TODO put the logic for getting field parameters into a function and add per field options
                     image_method = index_info.index_settings[NsField.index_defaults][NsField.image_preprocessing][NsField.patch_method]
                     # the chunk_image contains the no-op logic as of now - method = None will be a no-op
-                    content_chunks, text_chunks = image_processor.chunk_image(field_content, 
-                                    device=selected_device,
-                                    method=image_method)            
+                    content_chunks, text_chunks = image_processor.chunk_image(
```
Comment (jn2clark) [inline src/marqo/tensor_search/tensor_search.py]: the error will also need to be caught here. if an image is chunked then there will be an attempt to retrieve it here prior to the vectorise call

**Review by jn2clark — COMMENTED** [review]
I left another comment but the ```try``` for catching the error will also need to encompass the chunking as well

File: src/marqo/tensor_search/tensor_search.py (hunk 1 above)
Comment (pandu-k) [reply src/marqo/tensor_search/tensor_search.py]: added the try except block around chunking and tested

Comment (jn2clark) [reply src/marqo/tensor_search/tensor_search.py]: Thank you!

**Review by jn2clark — APPROVED** [review]

**Pull request merged**